expression/builtin_cast.go

```diff
@@ -107,6 +115,53 @@ var (
 	_ builtinFunc = &builtinCastJSONAsJSONSig{}
 )
 
+type castAsYearFunctionClass struct {
+	baseFunctionClass
+
+	tp *types.FieldType
+}
+
+//TODO: Set correct PbCode for function signature after `cast as year` related pb code are supported
+func (c *castAsYearFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (sig builtinFunc, err error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	bf := newBaseBuiltinCastFunc(newBaseBuiltinFunc(ctx, args), ctx.Value(inUnionCastContext) != nil)
+	bf.tp = c.tp
+	if args[0].GetType().Hybrid() || IsBinaryLiteral(args[0]) {
+		sig = &builtinCastIntAsIntSig{bf}
+		sig.setPbCode(tipb.ScalarFuncSig_CastIntAsInt)
+		return sig, nil
+	}
+	argTp := args[0].GetType().EvalType()
+	switch argTp {
+	case types.ETInt:
+		sig = &builtinCastIntAsYearSig{bf, &builtinCastIntAsIntSig{bf}}
+		sig.setPbCode(tipb.ScalarFuncSig_CastIntAsInt)
```

Comment:
it should be `tipb.ScalarFuncSig_CastIntAsYear`?

Reply:
@zz-jason we should add function signature like `tipb.ScalarFuncSig_CastIntAsYear` in `tipb`. Before that, I uses `tipb.ScalarFuncSig_CastIntAsInt` instead. 
Maybe we should not set signature before  that's done ?

Reply:
@zz-jason I've fixed the pb code. This depends on https://github.com/pingcap/tipb/pull/114